### PR TITLE
NEW Add isLiveVersion and isLatestDraftVersion to Versioned and GraphQL DataObject scaffolding

### PR DESCRIPTION
**Pull request by robbieaverill**
This surfaces whether each version is the current live version and the current/latest draft or modified version.

Required for #37

### CODE REVIEW

File: src/Versioned.php [hunk 1]
```diff
@@ -2379,7 +2379,7 @@ public function isLiveVersion()
         }
 
         $liveVersionNumber = static::get_versionnumber_by_stage($this->owner, Versioned::LIVE, $id);
-        return $liveVersionNumber === $this->owner->Version;
+        return (int) $liveVersionNumber === (int) $this->owner->Version;
```
Comment (dhensby) [inline src/Versioned.php]: I'm not sure of the benefit of strict type checking with casting? we might as well do a loose check?

Comment (robbieaverill) [reply src/Versioned.php]: Updated, cheers

**Comment by robbieaverill** [comment]
Rebased to get tests passing

**Comment by robbieaverill** [comment]
Self merging with three approvals plus @dhensby's comments addressed

**Pull request merged**